### ISSUE TITLE
Empty output caused by no reported overlap in annotation

### ISSUE THREAD

**Issue by M-K1** [mlbendall/telescope#45]
Hello @mlbendall,

I've been trying to work with your tool for some time now but can't seem to get it working outside of the test data you provided.
When running telescope on my own data and following the steps you provide, the tool reports 0 fragments overlapping with the annotation. The annotation I'm using is also the one you provided, namely the transcript.gtf file from [HERV_rmsk.hg38.v2](https://github.com/mlbendall/telescope_annotation_db/tree/master/builds/HERV_rmsk.hg38.v2). 

I inspected the mapping of the reads using IGV and saw that there are a fair amount of reads mapping within the regions of annotation, albeit with less coverage compared to the test data. That's why I am asking you if you can help me figure out why nothing is reported. I used my name sorted bam files and the aforementioned gtf file as only options when starting telescope assign.

Parameters used:
![image](https://github.com/mlbendall/telescope/assets/63231126/689c86d9-466e-4043-81a3-2c9bfbaf8263)

Logs of one of the samples:
![image](https://github.com/mlbendall/telescope/assets/63231126/20be6110-d0b0-45aa-b8b7-c8bfb79769d8)

Besides that, when starting a run I get the following warning, but I don't know whether this may be the cause of my problem:
`.conda/envs/telescope/lib/python3.7/importlib/_bootstrap.py:219: RuntimeWarning: pysam.libcalignedsegment.PileupColumn size changed, may indicate binary incompatibility. Expected 72 from C header, got 88 from PyObject`

Thanks in advance,

Matthijs

**Comment by sanhe374** [comment]
Hello, 
I have a similar issue and would like to know how to solve this issue.

**Comment by M-K1** [comment]
For those interested,

I figured out what the cause is for telescope reporting no overlap, which is caused by the annotation and headers of your bam file being different. The HERV_rmsk_hg38_v2 file has it's chromosomes listed as  "chr1, chr2, ..., chrX, chrY" and if this isn't the case in your bam files, the error occurs. 

You can change your headers the following way (with thanks to this [biostars](https://www.biostars.org/p/334717/#335343) post):

```
samtools view -H your_file.bam > header
...edit the header in your favourite text editor ...
samtools reheader header your_file.bam > fixed.bam
```

It would be nice if telescope could have a built in function for reading and converting the standard annotations of chromosomes into the annotations present in HERV_rmsk_hg38_v2 during telescope assign, since the genome I am using is from Ensembl, which you would expect to be a suitable reference genome.

**Comment by sanhe374** [comment]
Hello,

Thank you!
I think you forgot to write how to change the header.

Kind regards,

Sandra

Skickat från Outlook för iOS<https://aka.ms/o0ukef>
________________________________
Från: M-K1 ***@***.***>
Skickat: Wednesday, August 30, 2023 11:58:26 AM
Till: mlbendall/telescope ***@***.***>
Kopia: Sandra Hellberg ***@***.***>; Comment ***@***.***>
Ämne: Re: [mlbendall/telescope] Empty output caused by no reported overlap in annotation (Issue #45)


Closed #45<https://github.com/mlbendall/telescope/issues/45> as completed.

**Comment by M-K1** [comment]
missclicked before I finished the comment. It should be complete now.
